x-pack/metricbeat/module/azure/azure.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azure
+
+import (
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/pkg/errors"
+	"time"
+)
+
+// Config options
+type Config struct {
+	ClientID            string           `config:"client_id"    validate:"required"`
+	ClientSecret        string           `config:"client_secret" validate:"required"`
+	TenantID            string           `config:"tenant_id" validate:"required"`
+	SubscriptionID      string           `config:"subscription_id" validate:"required"`
+	Period              time.Duration    `config:"period"`
```

Comment:
Does it get the default period if it is not set in the configuration? Another way of obtaining the period that is actually being used by the metricset would be to get it from `base.Module().Config().Period`.